homeassistant/helpers/entity_registry.py

```diff
@@ -344,6 +344,10 @@ def async_get_or_create(
         entity_id = self.async_get_entity_id(domain, platform, unique_id)
 
         if entity_id:
+            # Don't change has_entity_name if a name has been set already
+            if has_entity_name and self.entities[entity_id].name is not None:
+                has_entity_name = None
+
```

Comment:
This is not correct, it will prevent integrations from upgrading their entities.